### DISCUSSION TITLE
Route to a URL entirely with query parameters?

### DISCUSSION THREAD

**Discussion by krileon** [symfony/symfony#48989] [Q&A] [answered]
I'm working within an environment where URL rewriting might not be available, but I still need to match a route. Is the only way to process the query parameters on / root route or can a REGEXP be defined to match the query parameters as well instead? Basically a nested symfony skeleton inside of another system is what I'm doing. A URL might be something like the following.

index.php?source=component&view=articles.list&category=1

So far have only had success parsing the query parameters out in / and hardcoding in the routes there based off the query parameters.

**Comment by MatTheCat** [comment, marked as answer]
> The query string of a URL is not considered when matching routes. https://symfony.com/doc/current/routing.html

([Conditions](https://symfony.com/doc/current/routing.html#matching-expressions) wouldn’t help as they can only **restrict** the matching.)

So what you’re doing sounds like a viable approach. Not sure you need a “nested” app though: you could [forward requests](https://symfony.com/doc/current/controller/forwarding.html) in the `/` controller.

**Reply by krileon** [reply]
Thank you, good to know I've basically been on the right track! I suppose the next thing to try then is to see if I can extend the router and implement it in the matcher or maybe try to implement it as middleware using the event dispatcher.

> Not sure you need a “nested” app though: you could [forward requests](https://symfony.com/doc/current/controller/forwarding.html) in the / controller.

It seams silly to be nested in something else, but it's basically a plugin in another CMS.
